src/test/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoffTest.java

```diff
@@ -0,0 +1,278 @@
+
+        assertThat(decoratedSphereClient.execute(customerUpdateCommand))
+            .failsWithin(4, TimeUnit.SECONDS)
+            .withThrowableOfType(ExecutionException.class)
+            .withCauseExactlyInstanceOf(GatewayTimeoutException.class)
+            .withMessageContaining("504");
+
+        verify(mockSphereUnderlyingClient, times(4)).execute(customerUpdateCommand);
+    }
+
+    @Test
+    void of_withRetryDecorator_ShouldNotRetryWhen400HttpResponse() {
+        final SphereClientConfig clientConfig =
+                SphereClientConfig.of("project-key", "client-id", "client-secret");
+        final RetryableSphereClientWithExponentialBackoff retryableSphereClientWithExponentialBackoff =
+                RetryableSphereClientWithExponentialBackoff.of(clientConfig);
+        final SphereClient mockSphereUnderlyingClient =
+            spy(createHttpTestDouble(intent -> HttpResponse.of(HttpStatusCode.BAD_REQUEST_400,
+                "{\"statusCode\":\"400\"}")));
+
+        final long maxRetryAttempt = 2L;
+        final Function<RetryContext, Duration> durationFunction = retryContext -> Duration.ofSeconds(2);
+
+        final SphereClient decoratedSphereClient = retryableSphereClientWithExponentialBackoff.decorateSphereClient(
+                mockSphereUnderlyingClient, maxRetryAttempt, durationFunction, DEFAULT_MAX_PARALLEL_REQUESTS);
+
+        final CustomerUpdateCommand customerUpdateCommand = getCustomerUpdateCommand();
+
+        assertThat(decoratedSphereClient.execute(customerUpdateCommand))
+            .failsWithin(1, TimeUnit.SECONDS)
+            .withThrowableOfType(ExecutionException.class)
+            .withCauseExactlyInstanceOf(ErrorResponseException.class)
+            .withMessageContaining("400");
+
+        // No retry, only first request.
+        verify(mockSphereUnderlyingClient, times(1)).execute(customerUpdateCommand);
+    }
+
+    @Test
+    void calculateExponentialRandomBackoff_withRetries_ShouldReturnRandomisedDurations() {
+        final SphereClientConfig clientConfig =
+                SphereClientConfig.of("project-key", "client-id", "client-secret");
+        final RetryableSphereClientWithExponentialBackoff retryableSphereClientWithExponentialBackoff =
+                RetryableSphereClientWithExponentialBackoff.of(clientConfig);
+        long maxDelay = 0;
+        for (long failedRetryAttempt = 1; failedRetryAttempt <= 10; failedRetryAttempt++) {
+
+            maxDelay += DEFAULT_INITIAL_RETRY_DELAY * ((long) Math.pow(2, failedRetryAttempt - 1)) * 2;
+
+            /* One example of wait times of retries:
+            Retry 1: 318 millisecond
+            Retry 2: 740 millisecond
+            Retry 3: 1284 millisecond
+            Retry 4: 2002 millisecond
+            Retry 5: 6054 millisecond
+            Retry 6: 8690 millisecond
+            Retry 7: 15567 millisecond
+            Retry 8: 30000 millisecond
```

Comment:
Can you update the examples, because max Delay is 60.000 now 😉

Reply:
Updated. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/2f7b7094cddbffeeb37ecf1577f79e0a606b216c